packages/mdl-base-component/adapter.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * A basic component adapter.
+ */
+export const MDLBaseAdapter = {
+  getBoundingClientRect(id) {
+    return this.elements_[id].getBoundingClientRect();
+  },
+
+  setStyles(id, styles) {
+    Object.assign(this.elements_[id].style, styles);
+  },
+
+  forceLayout(id) {
+    return this.elements_[id].offsetWidth;
+  },
+
+  addClass(id, cls) {
+    this.elements_[id].classList.add(cls);
+  },
+
+  removeClass(id, cls) {
+    this.elements_[id].classList.remove(cls);
+  },
+
+  get(...symbolPath) {
```

Comment:
I'm still not sure about this...I think it may be a bit _too_ general in that it could be confusing for implementers. Furthermore, there's no guarantee that what they're returning is a DOM element. In the case of angular universal for example, you're expected to use angular2's renderer to query/manipulate the DOM. 

I'm still leaning towards having explicit methods that operate on subelements, e.g. `addNativeInputListener`, `activateForegroundRipple(rippleIdx)`, etc.